homeassistant/components/azure_devops/sensor.py

```diff
@@ -1,126 +1,99 @@
-    except aiohttp.ClientError as exception:
-        _LOGGER.warning(exception)
-        raise PlatformNotReady from exception
-
-    for build in builds:
-        sensors.append(
-            AzureDevOpsLatestBuildSensor(client, organization, project, build)
+    coordinator: DataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    _, _, builds = coordinator.data
+
+    sensors = [
+        AzureDevOpsSensor(
+            coordinator,
+            AzureDevOpsSensorEntityDescription(
+                key=f"{build.project.id}_{build.definition.id}_latest_build",
+                name=f"{build.project.name} {build.definition.name} Latest Build",
+                icon="mdi:pipe",
+                attrs=lambda build: {
+                    "definition_id": build.definition.id,
+                    "definition_name": build.definition.name,
+                    "id": build.id,
+                    "reason": build.reason,
+                    "result": build.result,
+                    "source_branch": build.source_branch,
+                    "source_version": build.source_version,
+                    "status": build.status,
+                    "url": build.links.web,
+                    "queue_time": build.queue_time,
+                    "start_time": build.start_time,
+                    "finish_time": build.finish_time,
+                },
+                build_key=key,
+                organization=entry.data[CONF_ORG],
+                value=lambda build: build.build_number,
+            ),
         )
+        for key, build in enumerate(builds)
+    ]
 
     async_add_entities(sensors, True)
 
 
 class AzureDevOpsSensor(AzureDevOpsDeviceEntity, SensorEntity):
-    """Defines a Azure DevOps sensor."""
+    """Define a Azure DevOps sensor."""
 
     def __init__(
         self,
-        client: DevOpsClient,
-        organization: str,
-        project: str,
-        key: str,
-        name: str,
-        icon: str,
-        measurement: str = "",
-        unit_of_measurement: str = "",
-    ) -> None:
-        """Initialize Azure DevOps sensor."""
-        self._attr_native_unit_of_measurement = unit_of_measurement
-        self.client = client
-        self.organization = organization
-        self.project = project
-        self._attr_unique_id = "_".join([organization, key])
-
-        super().__init__(organization, project, name, icon)
-
-
-class AzureDevOpsLatestBuildSensor(AzureDevOpsSensor):
-    """Defines a Azure DevOps card count sensor."""
-
-    def __init__(
-        self, client: DevOpsClient, organization: str, project: str, build: DevOpsBuild
+        coordinator: DataUpdateCoordinator,
+        description: AzureDevOpsSensorEntityDescription,
     ) -> None:
         """Initialize Azure DevOps sensor."""
-        self.build: DevOpsBuild = build
+        self._attr_unique_id = "_".join([description.organization, description.key])
+        self.entity_description: AzureDevOpsSensorEntityDescription = description
```

Comment:
These lines should be moved to `AzureDevOpsEntity`
When you do that, you can remove the init method of `AzureDevOpsSensor`

Reply:
Had to leave the init in to avoid the lint error that value and attrs don't exist on SensorDescription